src/tree.rs

```diff
@@ -436,6 +438,63 @@ impl<T> Tree<T> {
     }
 
     ///
+    /// Swaps two `Node`s including their children given their `NodeId`s.
+    ///
+    /// Returns an empty `Result` containing a `NodeIdError` if one occurred on either provided `Id`.
+    ///
+    /// Notes:
+    ///     - Both `NodeId`s are still valid after this process and are not swapped.
+    ///     - Keeps the positions of the `Node`s in their parents children collection.
+    ///
+    /// ``
```

Comment:
@Drakulix haha, ooops, I had forgotten that we agreed already on Option 2.  My bad!

And I agree actually, Option 2's behavior is probably what most people would expect here.

With those rules in mind, I just noticed that (unless I'm missing an edge case) if we're executing an up/down swap, the upper `Node` will always end up as the last child of the lower `Node`.  Is that right?